docs/next.config.js

```diff
@@ -58,6 +58,7 @@ const nextConfig = withNextra({
   },
   reactStrictMode: true,
   experimental: {
+    newNextLinkBehavior: true,
```

Comment:
This switches us over to the "no `<a>` required" mode while still on Next.js 12.

Lots of changes as a consequence of this.